decidim-core/spec/commands/decidim/search_spec.rb

```diff
@@ -0,0 +1,249 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim
+  describe Search do
+    describe "call" do
+      let(:current_organization) { create(:organization) }
+
+      context "with resources from different organizations" do
+        let(:other_organization) { create(:organization) }
+        let(:term) { "fire" }
+
+        before do
+          create(:searchable_resource, organization: current_organization, content_a: "Fight fire with fire")
+          create(:searchable_resource, organization: other_organization, content_a: "Light my fire")
+        end
+
+        it "returns resources only from current_organization" do
+          described_class.call(term, current_organization) do
+            on(:ok) do |results|
+              expect(results.count).to eq(1)
+              expect(results.first.organization).to eq(current_organization)
+            end
+            on(:invalid) { raise("Should not happen") }
+          end
+        end
+      end
+
+      context "with resources indexed in many languages" do
+        let!(:lice_ca) { create(:searchable_resource, organization: current_organization, locale: :ca, content_a: "Erradicació de polls a l'escola") }
+        let!(:lice_en) { create(:searchable_resource, organization: current_organization, locale: :en, content_a: "Eradication of lice in school") }
+        let!(:ci_ca) { create(:searchable_resource, organization: current_organization, locale: :ca, content_a: "Millora continua mitjançant enquestes periòdiques") }
+        let!(:ci_en) { create(:searchable_resource, organization: current_organization, locale: :en, content_a: "Continous improvement with periodic polls") }
+
+        context "when term has matches in many languages" do
+          let(:term) { "polls" }
+          let(:locale_before) { I18n.locale }
+
+          before do
+            I18n.locale = :ca
+          end
+
+          it "returns only results in current language" do
+            begin
+              described_class.call(term, current_organization) do
+                on(:ok) do |results|
+                  expect(results.pluck(:id)).to eq([lice_ca.id])
+                end
+                on(:invalid) { raise("Should not happen") }
+              end
+            ensure
```

Comment:
What about using `before` and `after` blocks?

You could also wrap the call with `I18n.with_locale(:ca) do`